src/sfincs_bmi.cxx

```diff
@@ -11,239 +11,349 @@
 
 #include <cstring>
 #include <iostream>
+
 #include "sfincs_bmi.hxx"
 
 // sfincs_bmi.f90 implements/exports the following bmi functions:
-extern "C" int initialize(char *c_config_file);
-extern "C" int update(double dt);  // Doesn't seem to update current time
+extern "C" int initialize(const char *c_config_file);
 extern "C" int finalize();
-extern "C" void get_start_time(double *tstart);
-extern "C" void get_end_time(double *tend);
-extern "C" void get_time_step(double *dt);
-extern "C" void get_current_time(double *tcurrent);
-extern "C" void get_var_type(const char *c_var_name, char *c_type);
+
+extern "C" int get_component_name(char *name);
+
+extern "C" int update();
+extern "C" int update_until(double t);
+
+extern "C" int get_start_time(double *tstart);
+extern "C" int get_end_time(double *tend);
+extern "C" int get_time_step(double *dt);
+extern "C" int get_time_units(char *units);
+extern "C" int get_current_time(double *tcurrent);
+
+extern "C" int get_var_type(const char *c_var_name, char *c_type);
 
 // These function are also exported but not in the BMI spec
-extern "C" void get_var(const char *c_var_name, void *x);  // not exported; should be get_value?
-extern "C" void set_var(const char *c_var_name, float *xptr);  // should be set get_value?
-extern "C" void get_var_shape(const char *c_var_name, int *var_shape); // should be get_grid_shape
-// extern "C" void get_var_rank(char *c_var_name, int *rank); // should be get_grid_rank
+extern "C" int get_value(const char *c_var_name, void *x); // not exported; should be get_value?
```

Comment:
Doesn't seem to work like before, or as expected. Calling with 
```py
grid_id = 0
grid_rank = model.get_grid_rank(grid_id)
shape = model.get_grid_shape(grid_id, np.empty(grid_rank, dtype=int))

model.get_value('zs', np.zeros(shape[0]))
# or
model.get_value('zs', np.zeros(shape[1]))
# or
model.get_value('zs', np.zeros(886*908))
```
gives segmentation fault. 
What is the shape supposed to be? 

```
model.get_grid_shape(grid_id, np.empty(grid_rank, dtype=int))
```
gives `array([313509,  32614]`, that seems very big!! In sfincs.ini, I have 

```
mmax                 = 886
nmax                 = 908
```

Reply:
Conclusion: the second number is meaningless, first number is total points. It is less than nmax*mmax, I guess because some points in the grid are masked/boundaries/...